cci_cameroon/analysis/text_similarity/similarity_search_sample.py

```diff
@@ -0,0 +1,608 @@
+sample_french2 = ["C'est une personne très heureuse"]
+
+# %%
+print(torch.__version__)
+
+# %% [markdown]
+# ## 1. https://www.pinecone.io/learn/faiss-tutorial/
+# 2. Choosing index https://towardsdatascience.com/billion-scale-semantic-similarity-search-with-faiss-sbert-c845614962e2
+# 3. Points to consider when choosing an index: https://github.com/facebookresearch/faiss/wiki/Guidelines-to-choose-an-index
+# 4. distilBert impl mis-info covid  https://github.com/kstathou/vector_engine/blob/master/notebooks/001_vector_search.ipynb
+#
+# 5. Roberta https://pytorch.org/hub/pytorch_fairseq_roberta/
+#
+# 6. Camembert is French implementation of Roberta https://github.com/pytorch/fairseq/tree/main/examples/camembert
+# 7. Paper with multilingual text embeddings - frence : https://jep-taln2020.loria.fr/wp-content/uploads/JEP-TALN-RECITAL-2020_paper_209.pdf MUSE and M-BERT are chosen.
+#
+#
+#
+
+# %%
+# #!pip install bitarray
+# #!pip install hydra-core omegaconf
+# #!pip install fairseq
+# !pip install --no-cache-dir sentencepiece
+
+# %%
+import torch
+
+# workaround to 403 Http limit exceeded, include the following line of code before using torch.hub
+torch.hub._validate_not_a_forked_repo = lambda a, b, c: True
+xlmr = torch.hub.load("pytorch/fairseq:main", "xlmr.large")
+xlmr.eval()  # disable dropout (or leave in train mode to finetune)
+
+# %%
+print(xlmr)
+
+# %%
+print(embeddings11)
+
+# %% [markdown]
+# # FlauBert pre-trained model
+
+# %%
+model_flaubert = SentenceTransformer("hugorosen/flaubert_base_uncased-xnli-sts")
+
+
+# %%
+# create embeddings of our sample frence sentences
+embeddings_flaubert = model_flaubert.encode(sentences_french)
+embedding_sample1_fr = model_flaubert.encode(sample_french1)
+embedding_sample2_fr = model_flaubert.encode(sample_french2)
+
+# %%
+scores_flaubert1 = util.pytorch_cos_sim(embedding_sample1_fr, embeddings_flaubert)
+scores_flaubert2 = util.pytorch_cos_sim(embedding_sample2_fr, embeddings_flaubert)
+print(scores_flaubert1)
+print(scores_flaubert2)
+
+# %% [markdown]
+# # Using a pre-trained model for French text : french_semantic
+# This model is trained using text in french language. We evaluate its performance on french text below.
+# The difference in the models lies mostly at the level of embedding the text.
+
+# %%
+# using the french_semantic model for embedding
+model_fr = SentenceTransformer("Sahajtomar/french_semantic")
+
+# %%
+sentences1 = [
+    "J'aime mon téléphone",
+    "Mon téléphone n'est pas bon.",
+    "Votre téléphone portable est superbe.",
+]
+sentences2 = [
+    "Est-ce qu'il neige demain?",
+    "Récemment, de nombreux ouragans ont frappé les États-Unis",
+    "Le réchauffement climatique est réel",
+]
+embeddings11 = model_fr.encode(sentences1, convert_to_tensor=True)
+embeddings22 = model_fr.encode(sentences2, convert_to_tensor=True)
+cosine_scores = util.pytorch_cos_sim(embeddings11, embeddings22)
+for i in range(len(sentences1)):
```

Comment:
Can you convert this into a function to sit in pipeline

Reply:
Thinking that once we agree on which approach to use, the functions needed could then be written in the appropriate locations?